src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15589,7 +15589,54 @@ def pct_change(
                 ).frame
             )
 
-    def equals(self, other: "SnowflakeQueryCompiler") -> "SnowflakeQueryCompiler":
+    def index_equals(self, other: "SnowflakeQueryCompiler") -> bool:
+        """
+        Compare self index against other index.
+        The things that are being compared are:
+        * The elements inside the Index object.
+        * The order of the elements inside the Index object.
+
+        Args:
+            other: Snowflake query compiler to compare against.
+
+        Returns:
+            True if self index is equal to other index, False otherwise.
+        """
+        if self._modin_frame.num_index_columns != other._modin_frame.num_index_columns:
+            return False
```

Comment:
It seems we need a test covering this case when the number of index columns is not matching.

Reply:
Ah, this needs to go away. I later learned that currently we only support lazy index for single index.